src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -387,12 +387,31 @@ private void resolveSampleParamConfigs(
       fieldPath = parentFieldPath + "." + key;
     }
     this.sampleParamConfig = sampleParamConfigMap.get(fieldPath);
-    if (sampleParamConfig != null && sampleParamConfig.readFromFile()) {
-      setLineType(InitCodeLineType.ReadFileInitLine);
+
+    if (sampleParamConfig != null) {
+
+      // If read_file is set to true in config, change the line type to ReadFileInitLine.
+      if (sampleParamConfig.readFromFile()) {
+        setLineType(InitCodeLineType.ReadFileInitLine);
+      }
+
+      // If sample_argument_name is specified in config, set identifier to this name if the name has
+      // not be used yet and error out otherwise.
+      if (sampleParamConfig.isSampleArgument()) {
+        Name argName = Name.anyLower(sampleParamConfig.sampleArgumentName());
+        if (!argName.equals(identifier)) {
+          Preconditions.checkArgument(
+              !context.symbolTable().isSymbolUsed(argName),
+              "sample_argument_name \"%s\" is not valid.",
+              sampleParamConfig.sampleArgumentName());
+          identifier =
+              context
+                  .symbolTable()
+                  .getNewSymbol(Name.anyLower(sampleParamConfig.sampleArgumentName()));
+        }
+      }
     }
-    children
-        .values()
-        .forEach(child -> child.resolveSampleParamConfigs(sampleParamConfigMap, fieldPath));
+    children.values().forEach(child -> child.resolveSampleParamConfigs(context, fieldPath));
```

Comment:
Can it ever happen that `sampleParamConfig != null` above and then we recurse here? I would expect that we only recurse if `sampleParamConfig == null`. Is that correct? If so, an assertion would be good. If not, I'd like to understand this better.

Reply:
It might be possible as we might specify the `sample_argument_name` (or `sample_argument`) for both a parent node and one of it's children. It's never used for now but should be a valid use case. For example:
```
atrributes:
- parameter: foo
  sample_argument_name: foo_name
- parameter: foo.bar
  sample_argument_name: foo_bar_name
```

Reply:
But does that make sense? They are overlapping specifications. What are the semantics? Should the latter one override the former one? Should the more specific one override the more general one?

If our code doesn't have set semantics for this situation, or if we don't have a way of dealing with such overlap (including tests!), I think it's safer to explicitly disallow it; we can deliberately add it later if needed.

Reply:
Oh I was wrong! I confused `sample_argument_name` with `variable_name` :/ Yeah I think you are right that it would never happen. I'll add a check there. And actually we do have [assertions](https://github.com/googleapis/gapic-generator/blob/e052651a59cf6e5a12c1f4c77dcc58d17ecdd670/src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java#L246).

Reply:
@vchudnov-g It seems repetitive to do what `assertNoOverlap` again here, since we are going to do this check after the tree is built. Are you OK if I just skip the recursion if `sampleParamConfig != null`? I've moved the recursion into an `else` block.

Reply:
sgtm. I'd recommend adding a brief comment making that we're purposefully not recursing since overlaps are not allowed.